pkgs/test/nixpkgs-check-by-name/src/eval.nix

```diff
@@ -79,15 +79,34 @@ let
         };
       };
 
-  attrInfos = map (name: [
+  byNameAttrs = map (name: [
     name
-    (
-      if ! pkgs ? ${name} then
-        { Missing = null; }
-      else
-        { Existing = attrInfo name pkgs.${name}; }
-    )
+    {
+      ByName =
+        if ! pkgs ? ${name} then
+          { Missing = null; }
+        else
+          { Existing = attrInfo name pkgs.${name}; };
+    }
   ]) attrs;
 
+  # Information on all attributes that exist but are not in pkgs/by-name.
+  # We need this to enforce pkgs/by-name for new packages
+  nonByNameAttrs = map (name:
+    let
+      output = attrInfo name pkgs.${name};
+      result = builtins.tryEval (builtins.deepSeq output null);
+    in
+    [
+      name
+      {
+        NonByName =
+          if result.success then
+            { EvalSuccess = output; }
+          else
+            { EvalFailure = null; };
+      }
+    ]
+  ) (builtins.attrNames (builtins.removeAttrs pkgs attrs));
 in
-attrInfos
+byNameAttrs ++ nonByNameAttrs
```

Comment:
I don't think this is sorted anymore. Might not matter.

Reply:
Doesn't matter much, but it makes the resulting error ordering a bit nicer, I adjusted it now to always sort it in https://github.com/NixOS/nixpkgs/pull/275539/commits/cbff02834f3c0a1daa3502566fcf19433f9ca214 :)